### ISSUE TITLE
Proper way to get regular pcode from raw pcode

### ISSUE THREAD

**Issue by jstaursky** [NationalSecurityAgency/ghidra#1215]
So I am playing with the example given [here](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Decompiler/src/decompile/cpp/sleighexample.cc). For the sleigh library.

and in particular I want to show the register names attached to the varnodes in the raw pcode output.

to that end I modified `dumpPcode` as such

```cpp
static void dumpPcode(Sleigh &trans)

{ // Dump pcode translation of machine instructions
  PcodeRawOut emit;		// Set up the pcode dumper
  AssemblyRaw assememit;	// Set up the disassembly dumper
  int4 length;			// Number of bytes of each machine instruction

  Address addr(trans.getDefaultSpace(),0x80483b4);     // First address to translate
  Address lastaddr(trans.getDefaultSpace(),0x80483bf); // Last address

  while(addr < lastaddr) {
    cout << "--- ";
    trans.printAssembly(assememit,addr);
    length = trans.oneInstruction(emit,addr); // Translate instruction
    // *** MODIFIED HERE ******************************************
    string regname =
        trans.getRegisterName(addr.getSpace(), addr.getOffset(), addr.getAddrSize());
    cout << regname << endl;
    // *** END MODIFIED ******************************************
    addr = addr + length;		// Advance to next instruction
  }
}
```

However this doesn't seem to have any affect. I know that `PcodeRawOut emit` is used to emit p-code when passed onto `trans.oneInstruction` but neither `PcodeRawOut` nor its base class `PcodeEmit` hold any methods to get the register name.

In short, **how do you get regular p-code** (populated with register names + flag names, etc) **from raw p-code**?

**Comment by jstaursky** [comment]
Here is how I eventually did it for anyone else who got stuck.
```cpp
static void print_vardata (ostream& s, VarnodeData& data)
{
    const Translate *trans = data.space->getTrans();

    s << '(' << data.space->getName() << ',';
    if (data.space->getName() == "register") {
        s << trans->getRegisterName(data.space, data.offset, data.size);
    } else {
        data.space->printOffset (s, data.offset);
    }
    s << ',' << dec << data.size << ')';
}


```
